nomad/rpc_test.go

```diff
@@ -1061,16 +1062,23 @@ func TestRPC_TLS_Enforcement_Raft(t *testing.T) {
 			cn:      "nomad.example.com",
 			canRaft: false,
 		},
+		// Support wildcard DNS names for backwards compatibility.
 		{
-			name:    "globs",
+			name:     "dns name globs",
+			cn:       "server.other.nomad",
+			dnsNames: []string{"*.global.nomad"},
+			canRaft:  true,
+		},
+		{
+			name:    "globs without role",
 			cn:      "*.global.nomad",
 			canRaft: false,
```

Comment:
Why doesn't this work? A CN of `*.global.noamd` should satisfy the `server.global.nomad` requirement unless CN's never support wildcards?